### PR TITLE
Added test if source is declare revocation and target is undeclared, declared, progressed and revoked

### LINKED ISSUE #418: Restrict progressOutboxWithProof in MessageBus when outbox state of source chain is `RevocationDeclared` and inbox state of target chain is `Declared`.
## User Story
If the message (Stake/Redeem) is *not* already progressed then Staker/Redeemer can revoke stake/redeem process.
There is no restriction as of now and anyone can progressOutboxWithProof and complete the process when the inbox message state is `Declared`. 
This is not the message initiator (Staker or Redeemer) intended and this will result into penalty loss.
Message initiator must have a complete control in such a scenario.

## Info
- While executing `progressOutboxWithProof` check the inbox message states. 
- If inbox message state is `Declared` and outbox message state is `RevocationDeclared` then `progressOutboxWithProof` should fail in this case.

We can also discuss if this can be allowed if the msg.sender is the message initiator.

### PR DESCRIPTION
**Pull request by sarvesh-ost**
Fixes #418  

Added missing test for progress outbox with proof.

### CODE REVIEW

File: test/lib/messagebus/progress_outbox_with_proof.js [hunk 1]
```diff
@@ -19,107 +19,114 @@
 //
 // ----------------------------------------------------------------------------
 
-const messageBusUtilsKlass = require('./messagebus_utils'),
-  messageBusUtils = new messageBusUtilsKlass();
+const messageBusUtilsKlass = require('./messagebus_utils');
```
Comment (schemar) [inline test/lib/messagebus/progress_outbox_with_proof.js]: Usually, we have classes uppercase and instances lowercase. E.g.:
```js
const MessageBusUtils = require('./messagebus_utils');
const messageBusUtils = new MessageBusUtils();
```

File: test/lib/messagebus/progress_outbox_with_proof.js [hunk 2]
```diff
@@ -19,107 +19,114 @@
...
+      + ' status at source is progressed', async () => {
+        await messageBusUtils.declareMessage(params, true);
+        params.messageStatus = MessageStatusEnum.Declared;
+        await messageBusUtils.progressOutboxWithProof(params, true);
+
+        params.messageStatus = MessageStatusEnum.Revoked;
+        await messageBusUtils.progressOutboxWithProof(params, false);
+    });
+
+    it('should fail when message status at target  is revoked and message status at source is'
```
Comment (deepesh-kn) [inline test/lib/messagebus/progress_outbox_with_proof.js]:
```suggestion
    it('should fail when message status at target is revoked and message status at source is'
```

File: test/lib/messagebus/progress_outbox_with_proof.js [hunk 3]
```diff
@@ -19,107 +19,114 @@
...
+        await messageBusUtils.declareRevocationMessage(params, true);
+        params.messageStatus = MessageStatusEnum.Revoked;
+        await messageBusUtils.progressOutboxRevocation(params, true);
+
+        params.messageStatus = MessageStatusEnum.Revoked;
+        await messageBusUtils.progressOutboxWithProof(params, false);
+    });
+
+
+    it('should fail when message status at target is empty and message status at '
```
Comment (deepesh-kn) [inline test/lib/messagebus/progress_outbox_with_proof.js]:
```suggestion
    it('should fail when message status at target is undeclared and message status at '
```

File: test/lib/messagebus/progress_outbox_with_proof.js [hunk 4]
```diff
@@ -19,107 +19,114 @@
...
-      
-      await messageBusUtils.declareMessage(params, true);
-      params.messageStatus = MessageStatusEnum.Declared;
-      await messageBusUtils.progressOutboxWithProof(params, true);
-      
-      params.messageStatus = MessageStatusEnum.Revoked;
-      await messageBusUtils.progressOutboxWithProof(params, false);
-      
+
+    it('should fail when message status at target  is empty and message status is'
```
Comment (deepesh-kn) [inline test/lib/messagebus/progress_outbox_with_proof.js]:
```suggestion
    it('should fail when message status at target is undeclared and message status is'
```

File: test/lib/messagebus/progress_outbox_with_proof.js [hunk 5]
```diff
@@ -19,107 +19,114 @@
...
-    it('should fail when message status at target is revoked and message status at source is declared revocation', async () => {
-      
-      await messageBusUtils.declareMessage(params, true);
-      await messageBusUtils.declareRevocationMessage(params, true);
-      
-      params.messageStatus = MessageStatusEnum.Revoked;
-      await messageBusUtils.progressOutboxWithProof(params, false);
-      
+
+    it('should fail when message status at target is empty and message'
```
Comment (deepesh-kn) [inline test/lib/messagebus/progress_outbox_with_proof.js]:
```suggestion
    it('should fail when message status at target is undeclared and message'
```

**Pull request merged**